### DISCUSSION TITLE
Easily allow Timeline item to take full width of container

### DISCUSSION THREAD

**Discussion by joelzwarrington** [mantinedev/mantine#2731] [Feature requests]
It would be nice to replicate the functionality of issue/PR discussion timelines with an item which takes the full width of the container and doesn't have a bullet.

As an example:
![image](https://user-images.githubusercontent.com/80279872/196496930-a6c75802-e2ae-46d0-ae9a-50ab010b7c81.png)

What I was able to do by modifying the CSS of [examples](https://mantine.dev/core/timeline/):
![image](https://user-images.githubusercontent.com/80279872/196496171-a7e1c786-bdd3-427a-8078-475cc9c94954.png)

**Comment by rtivital** [comment]
That's a very strange use case, I've seen anything like this before. I do not think that we should support it on library level if it is achievable with Styles API.

**Reply by joelzwarrington** [reply]
I think it's reasonable to want to take the full width of the container. It's visually appealing and help breaks up the flow of the content for particular timeline events (such as comments).

While I think it could be possible with the Styles API, it will be very hacky and more easily subject to breaking changes due to use of negative margins and not static values. If you're still adamant about this not being supported on the library level, no worries

**Reply by joelzwarrington** [reply]
@rtivital - I put up a draft PR here: https://github.com/mantinedev/mantine/pull/2744, which I think shows that the changes can be made entirely with the Styles API. But I feel like it's similar to the button `fullWidth` property and could have value to be supported at the library level.

Anyway let me know what you think, and if you do agree to implement such a feature and need help, let me know and would be happy to continue my PR.

**Reply by rtivital** [reply]
As I've said before, I do not see any reason to add this feature as there aren't any valid use cases for this.

**Reply by joelzwarrington** [reply]
I don't see what you consider a valid use case to be? Having a timeline item take full width appears to be a valid use case as I've shown UI examples that use this pattern?

**Reply by rtivital** [reply]
Alright, maybe I was not clear. I would rather not add this feature because:
- I've never seen it used on any of the websites
- You've provided an example that shows styles that were applied, not how it would look like in real application
- Maintaining the feature that has no real use outside of your specific use case is meaningless
